server/src/internalClusterTest/java/org/elasticsearch/snapshots/MultiClusterRepoAccessIT.java

```diff
@@ -0,0 +1,104 @@
+import java.util.Arrays;
+import java.util.function.Function;
+
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+public class MultiClusterRepoAccessIT extends AbstractSnapshotIntegTestCase {
+
+    private InternalTestCluster secondCluster;
+    private Path repoPath;
+
+    @Before
+    public void startSecondCluster() throws IOException, InterruptedException {
+        repoPath = randomRepoPath();
+        secondCluster = new InternalTestCluster(randomLong(), createTempDir(), true, true, 0,
+                0, "second_cluster", new NodeConfigurationSource() {
+            @Override
+            public Settings nodeSettings(int nodeOrdinal) {
+                return Settings.builder().put(MultiClusterRepoAccessIT.this.nodeSettings(nodeOrdinal))
+                        .put(NetworkModule.TRANSPORT_TYPE_KEY, getTestTransportType())
+                        .put(Environment.PATH_REPO_SETTING.getKey(), repoPath).build();
+            }
+
+            @Override
+            public Path nodeConfigPath(int nodeOrdinal) {
+                return null;
+            }
+        }, 0, "leader", Arrays.asList(ESIntegTestCase.TestSeedPlugin.class,
+                MockHttpTransport.TestPlugin.class, MockTransportService.TestPlugin.class,
+                MockNioTransportPlugin.class, InternalSettingsPlugin.class, MockRepository.Plugin.class), Function.identity());
+        secondCluster.beforeTest(random());
+    }
+
+    @After
+    public void stopSecondCluster() throws IOException {
+        IOUtils.close(secondCluster);
+    }
+
+    public void testConcurrentDeleteFromOtherCluster() throws InterruptedException {
+        internalCluster().startMasterOnlyNode();
+        internalCluster().startDataOnlyNodes(1);
+        final String repoNameOnFirstCluster = "test-repo";
+        final String repoNameOnSecondCluster = randomBoolean() ? "test-repo" : "other-repo";
+        createRepository(repoNameOnFirstCluster, "fs", repoPath);
+
+        secondCluster.startMasterOnlyNode();
+        secondCluster.startDataOnlyNode();
+        secondCluster.client().admin().cluster().preparePutRepository(repoNameOnSecondCluster).setType("fs")
+                .setSettings(Settings.builder().put("location", repoPath)).get();
+
+        createIndexWithRandomDocs("test-idx-1", randomIntBetween(1, 100));
+        createFullSnapshot(repoNameOnFirstCluster, "snap-1");
+        createIndexWithRandomDocs("test-idx-2", randomIntBetween(1, 100));
+        createFullSnapshot(repoNameOnFirstCluster, "snap-2");
+        createIndexWithRandomDocs("test-idx-3", randomIntBetween(1, 100));
+        createFullSnapshot(repoNameOnFirstCluster, "snap-3");
+
+        secondCluster.client().admin().cluster().prepareDeleteSnapshot(repoNameOnSecondCluster, "snap-1").get();
+        secondCluster.client().admin().cluster().prepareDeleteSnapshot(repoNameOnSecondCluster, "snap-2").get();
+
+        expectThrows(SnapshotException.class, () ->
```

Comment:
can we assert that the message here is something about concurrent repo access?

Reply:
Sure, done :)